docs/source/pages/tutorials/provenance.rst

```diff
@@ -34,46 +36,111 @@ Skip this section if you already know how to install Macaron.
+
+.. _fig_toga_osi_api:
+
+.. figure:: ../../_static/images/tutorial_osi_toga.png
+   :alt: Open Source Insight's API result for toga package
+   :align: center
+
+This image shows the attestation URL found in the Open Source Insight API result.
+
+By using the Open Source Insights API, Macaron can check that the discovered provenance is verified, as well as being a valid match of the user provided PURL. For this we can rely on the check results produced from the analysis run. In particular, we want to know the results of three checks: ``mcn_provenance_derived_repo_1``, ``mcn_provenance_derived_commit_1``, and ``mcn_provenance_verified_1``. The first two to ensure that the commit and the repository being analyzed match those found in the provenance file, and the last check to ensure that the provenance file has been verified.
+
+.. _fig_toga_pypi_checks:
+
+.. figure:: ../../_static/images/tutorial_toga_pypi.png
+   :alt: HTML report for ``toga 0.5.1``, summary
+   :align: center
+
+All three checks show they have passed, meaning Macaron has discovered the correct provenance for the user provided PURL, and determined that it is verified. To access the full report use the following:
+
+.. code-block:: shell
+
+  open output/reports/pypi/toga/toga.html
+
+***************************************
+Attestation Discovery for toga (GitHub)
+***************************************
+
+The toga library is interesting in that it has GitHub attestation or PyPI attestation depending on which particular version of it is analyzed. To discover a GitHub attestation, we can analyze version 0.4.8:
+
+.. code-block:: shell
+
+    ./run_macaron.sh analyze -purl pkg:pypi/toga@0.4.8
+
+During this analysis, Macaron will attempt to discover a GitHub attestation by computing the hash of the relevant artifact. This is a requirement of GitHub's API to view artifact attestation, see the `GitHub Attestation API <https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-attestations>`_. The hash is computed by downloading the artifact and analysing it with the SHA256 algorithm. With the hash, the GitHub API can be called to find the related attestation.
+
+In this particular case, the SHA256 hash of the toga 0.4.8 artifact is 0814a72abb0a9a5f22c32cc9479c55041ec30cdf4b12d73a0017aee58f9a1f00. A GitHub attestation can be found for this artifact `here <https://api.github.com/repos/beeware/toga/attestations/sha256:0814a72abb0a9a5f22c32cc9479c55041ec30cdf4b12d73a0017aee58f9a1f00>`_.
+
+Attestation discovered through GitHub cannot be ascertained as verified at this time. However, we can still be sure that the repository URL and commit digest associated with the user provided PURL match what is found within the attestation. This is reported by Macaron in two checks: ``mcn_provenance_derived_repo_1`` and ``mcn_provenance_derived_commit_1``.
```

Comment:
All attestations published on GitHub are signed, and their signatures are verified. Therefore, the verification check should pass for GitHub attestations, as long as we retrieve them directly from GitHub.